### PR TITLE
chore: update deps to github.com/zitadel/oidc/v3

### PR DESCRIPTION
**Pull request by rchincha**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/api/routes.go [hunk 1]
```diff
@@ -1873,22 +1873,22 @@ func (rh *RouteHandler) Logout(response http.ResponseWriter, request *http.Reque
 }
 
 // github Oauth2 CodeExchange callback.
-func (rh *RouteHandler) GithubCodeExchangeCallback() rp.CodeExchangeCallback {
+func GithubCodeExchangeCallback[C oidc.IDClaims](c *Controller) rp.CodeExchangeCallback[C] {
```
Comment (andaaron) [inline pkg/api/routes.go:1876]: Why was this change needed?

Comment (rchincha) [reply pkg/api/routes.go:1876]: zitadel v3 made tokens as a generic type, so this needs resolving.

Comment (rchincha) [reply pkg/api/routes.go:1876]: pkg/api/routes.go:1876:54: cannot use generic type rp.CodeExchangeCallback[C "github.com/zitadel/oidc/v3/pkg/oidc".IDClaims] without instantiation
pkg/api/routes.go:1878:11: cannot use generic type "github.com/zitadel/oidc/v3/pkg/oidc".Tokens[C "github.com/zitadel/oidc/v3/pkg/oidc".IDClaims] without instantiation
pkg/api/routes.go:1911:54: cannot use generic type rp.CodeExchangeUserinfoCallback[C "github.com/zitadel/oidc/v3/pkg/oidc".IDClaims, U rp.SubjectGetter] without instantiation
pkg/api/routes.go:1912:62: cannot use generic type "github.com/zitadel/oidc/v3/pkg/oidc".Tokens[C "github.com/zitadel/oidc/v3/pkg/oidc".IDClaims] without instantiation
pkg/api/routes.go:1915:17: info.GetEmail undefined (type "github.com/zitadel/oidc/v3/pkg/oidc".UserInfo has no field or method GetEmail)
pkg/api/routes.go:1925:19: info.GetClaim undefined (type "github.com/zitadel/oidc/v3/pkg/oidc".UserInfo has no field or method GetClaim)

**Comment by andaaron** [comment]
@rchincha https://github.com/project-zot/zot/pull/2585
